ui/selectors/selectors.js

```diff
@@ -1573,6 +1589,43 @@ export const getMemoizedUnapprovedTypedMessages = createDeepEqualSelector(
   (unapprovedTypedMessages) => unapprovedTypedMessages,
 );
 
+/**
+ * Get the display name for an address.
+ * This selector will look into the internal accounts and address book to find a display name for the address.
+ *
+ * @param {object} state - The Redux state object.
+ * @param {string} address - The address to get the display name for.
+ * @returns {string} The display name for the address.
+ */
+export const getAddressDisplayName = (state, address) => {
+  const {
+    address: caipAddress,
+    chain: { namespace, reference },
+  } = parseCaipAccountId(
+    isHexString(address) ? `eip155:1:${address}` : address,
+  );
+
+  const isEip155 = namespace === 'eip155';
+
+  const parsedAddress = isEip155
+    ? toChecksumHexAddress(caipAddress)
+    : caipAddress;
+
+  const accounts = getInternalAccounts(state);
+  const accountName = getAccountName(accounts, parsedAddress);
+
+  // Address book will only work for EVM accounts.
+  const addressBookEntry =
+    isEip155 &&
+    getAddressBookEntryByNetwork(
+      state,
+      parsedAddress,
+      `0x${decimalToHex(reference)}`,
+    );
+
+  return accountName || addressBookEntry?.name || shortenAddress(parsedAddress);
```

Comment:
```suggestion
  return accountName ?? addressBookEntry?.name ?? null;
```
I think the consuming component should decide how to show no name, instead of us forcing a short address